scope.go

```diff
@@ -166,23 +169,25 @@ func newRootScope(opts ScopeOptions, interval time.Duration) *scope {
 	}
 
 	s := &scope{
-		baseReporter:    baseReporter,
-		bucketCache:     newBucketCache(),
-		cachedReporter:  opts.CachedReporter,
-		counters:        make(map[string]*counter),
-		countersSlice:   make([]*counter, 0, _defaultInitialSliceSize),
-		defaultBuckets:  opts.DefaultBuckets,
-		done:            make(chan struct{}),
-		gauges:          make(map[string]*gauge),
-		gaugesSlice:     make([]*gauge, 0, _defaultInitialSliceSize),
-		histograms:      make(map[string]*histogram),
-		histogramsSlice: make([]*histogram, 0, _defaultInitialSliceSize),
-		prefix:          sanitizer.Name(opts.Prefix),
-		reporter:        opts.Reporter,
-		sanitizer:       sanitizer,
-		separator:       sanitizer.Name(opts.Separator),
-		timers:          make(map[string]*timer),
-		root:            true,
+		baseReporter:          baseReporter,
+		bucketCache:           newBucketCache(),
+		cachedReporter:        opts.CachedReporter,
+		counters:              make(map[string]*counter),
+		countersSlice:         make([]*counter, 0, _defaultInitialSliceSize),
+		defaultBuckets:        opts.DefaultBuckets,
+		done:                  make(chan struct{}),
+		gauges:                make(map[string]*gauge),
+		gaugesSlice:           make([]*gauge, 0, _defaultInitialSliceSize),
+		histograms:            make(map[string]*histogram),
+		histogramsSlice:       make([]*histogram, 0, _defaultInitialSliceSize),
+		prefix:                sanitizer.Name(opts.Prefix),
+		reporter:              opts.Reporter,
+		sanitizer:             sanitizer,
+		separator:             sanitizer.Name(opts.Separator),
+		timers:                make(map[string]*timer),
+		root:                  true,
+		counterChangeNotifyCh: make(chan *counter, 1024),
```

Comment:
hm few thoughts about this channel 
- any concerns about blocking once we hit it? 
- how did you arrive at this size?
- maybe this should be configurable too? (akin to https://github.com/uber-go/tally/blob/master/m3/reporter.go#L160 for eg)